satellite/console/consoleweb/consoleapi/referrals.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package consoleapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/private/post"
+	"storj.io/storj/satellite/console"
+	"storj.io/storj/satellite/console/consoleweb/consoleql"
+	"storj.io/storj/satellite/mailservice"
+	"storj.io/storj/satellite/referrals"
+)
+
+// ErrReferralsAPI - console referrals api error type.
+var ErrReferralsAPI = errs.Class("console referrals api error")
+
+// Referrals is an api controller that exposes all referrals functionality.
+type Referrals struct {
+	log                   *zap.Logger
+	service               *console.Service
+	referralsService      *referrals.Service
+	mailService           *mailservice.Service
+	ExternalAddress       string
+	LetUsKnowURL          string
+	TermsAndConditionsURL string
+	ContactInfoURL        string
+}
+
+// NewReferrals is a constructor for api referrals controller.
+func NewReferrals(log *zap.Logger, service *console.Service, referralsService *referrals.Service, mailService *mailservice.Service, externalAddress string, letUsKnowURL string, termsAndConditionsURL string, contactInfoURL string) *Referrals {
+	return &Referrals{
+		log:                   log,
+		service:               service,
+		referralsService:      referralsService,
+		mailService:           mailService,
+		ExternalAddress:       externalAddress,
+		LetUsKnowURL:          letUsKnowURL,
+		TermsAndConditionsURL: termsAndConditionsURL,
+		ContactInfoURL:        contactInfoURL,
+	}
+}
+
+// GetTokens returns referral tokens based on user ID.
+func (controller *Referrals) GetTokens(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	var tokensRequest struct {
+		UserID string `json:"userId"`
```

Comment:
Is it safe enough to pass userID as a parameter?
What if i'll send another user's ID? 
Does it make sense to make this endpoint to require auth? And take userID from auth token?